### PR TITLE
MGMT-3776 added ISO_IMAGE_TYPE env var

### PR DESCRIPTION
**Pull request by danielerez**
Added support for iso image type (full-iso/minimal-iso):
- Added ISO_IMAGE_TYPE env var to Makefile.
- Propagated iso_image_type to 'generate_and_download_image' funcs.

### CODE REVIEW

**Comment by carbonin** [comment]
This is waiting on https://github.com/openshift/assisted-service/pull/932, right?

**Comment by danielerez** [comment]
> This is waiting on [openshift/assisted-service#932](https://github.com/openshift/assisted-service/pull/932), right?

Right, pending on image_type in ImageCreateParams.

**Comment by danielerez** [comment]
> > This is waiting on [openshift/assisted-service#932](https://github.com/openshift/assisted-service/pull/932), right?
> 
> Right, pending on image_type in ImageCreateParams.

Ready for review now. @tsorya @YuviGold can you please take a look? :)

File: discovery-infra/test_infra/assisted_service_api.py [hunk 1]
```diff
@@ -131,9 +131,9 @@ def _download(self, response, file_path, verify_file_size=False):
                     f'Actual size: {actual_file_size}. Expected size: {content_length}'
                     )
 
-    def generate_image(self, cluster_id, ssh_key, static_ips=None):
+    def generate_image(self, cluster_id, ssh_key, image_type, static_ips=None):
```
Comment (YuviGold) [inline discovery-infra/test_infra/assisted_service_api.py:134]: Any certain reason why `generate_and_download_image` has `image_type` with a default value and `generate_image` doesn't? I think we could have the same signature (either both have a default or neither of them)

Comment (danielerez) [reply discovery-infra/test_infra/assisted_service_api.py:134]: Just cause generate_image is called only locally (in class), but sure can change to keep consistency.

**Review by YuviGold — APPROVED** [review]
LGTM

**Review by YuviGold — APPROVED** [review]

**Pull request merged**